WalkingChallenge/LoginViewController.swift

```diff
@@ -65,7 +65,13 @@ extension LoginViewController: FBSDKLoginButtonDelegate {
       alert(message: "Error logging in: \(error)", style: .cancel)
       return
     }
-    AppController.shared.login()
+    
+    if(FBSDKAccessToken.current() == nil){
```

Comment:
It's not part of the error hence it will never get into that block. Error object comes nil when "done" or "cancel" button is being tapped. I cleaned up the logic a little. Since its a Bool check and primitive type object, there is no need to put the guard on it.